src/common/hotbar-store.ts

```diff
@@ -0,0 +1,67 @@
+import { action, comparer, observable, toJS } from "mobx";
+import { BaseStore } from "./base-store";
+import migrations from "../migrations/hotbar-store";
+
+export interface HotbarItem {
+  entity: {
+    uid: string;
+  };
+  params?: {
+    [key: string]: string;
+  }
+}
+
+export interface Hotbar {
+  name: string;
+  items: HotbarItem[];
+}
+
+export interface HotbarStoreModel {
+  hotbars: Hotbar[];
+}
+
+export class HotbarStore extends BaseStore<HotbarStoreModel> {
+  @observable hotbars: Hotbar[] = [];
+
+  private constructor() {
+    super({
+      configName: "lens-hotbar-store",
+      accessPropertiesByDotNotation: false, // To make dots safe in cluster context names
+      syncOptions: {
+        equals: comparer.structural,
+      },
+      migrations,
+    });
+  }
+
+  @action protected async fromStore(data: Partial<HotbarStoreModel> = {}) {
+    this.hotbars = data.hotbars || [{
```

Comment:
If incoming `data` is partial is it correct to overwrite it fully?

Reply:
This only overwrites `hotbars` array ... what would be better syntax for "optional" data 🤔

Reply:
Also to add, why do we need a default hotbar? I thought hotbars were optional to use.